### ISSUE TITLE
QueryComposer does not exist when injecting QueryComposerWrapper

### ISSUE THREAD

**Issue by bandtank** [espocrm/espocrm#2872]
**Describe the bug**
This probably is not a bug, but it is a behavior that does not make sense. Due to how some classes are instantiated, it is not always possible to use the constructor to inject dependencies. The following code does not work:

```
use Espo\ORM\QueryComposer\QueryComposerWrapper;
...
$injct = $this->container->getByClass(InjectableFactory::class);
$queryComposer = $injct->create(QueryComposerWrapper::class);
```

Here is the error:
```
[2023-10-07 15:03:44] ERROR: Uncaught Exception RuntimeException: "InjectableFactory: Class 'Espo\ORM\QueryComposer\QueryComposer' does not exist."
```

**To Reproduce (MANDATORY, DO NOT REMOVE)**
Explicit steps to reproduce the behavior:
1. Run the code shown above in any class that can execute ORM code; e.g., a job or service
2. Inspect the log

**Expected behavior**
I expected the QueryComposer class to correctly inject \Espo\ORM\QueryComposer\QueryComposer.php. It fails because the QueryComposer.php file contains an **interface** instead of a class.

**EspoCRM version**
8.0.2

**Additional context**
This is happening in the AfterInstall.php script in the extension development template. I do not think that is relevant other than not being able to use the constructor for dependency injection.

# QUESTION
Can we please add a `toSql` method to the wrapper? It is frustrating sometimes to not be able to convert a query into SQL. I always modify the file to look like this:
```
class DefaultQueryExecutor implements QueryExecutor
{
    public function __construct(
        private SqlExecutor $sqlExecutor,
        private QueryComposerWrapper $queryComposer
    ) {}

    public function execute(Query $query): PDOStatement
    {
        $sql = $this->queryComposer->compose($query);

        return $this->sqlExecutor->execute($sql, true);
    }

    public function toSql(Query $query): string {
      return $this->queryComposer->compose($query);
    }
}
```

which allows me to do this:
```
$sql = $this->em->getQueryExecutor()->toSql($selectQuery);
```

I realize the composer can already perform this function, but the query executor figures out which composer to use, so it is an easier place to do the conversion.

**Comment by yurikuzn** [comment]
Not everything in the system is creatable via DI and it's fine. Some classes are behind the layer. In this case it's the EntityManager.

I don't understand the problem.

Why you don't use:

```json
$sql = $this->entityManager->getQueryComposer()->compose($query);
```

**Comment by bandtank** [comment]
I did not know it was possible to reach the query composer through the entity manager. Thank you for the explanation. I figured this wasn't a bug, but I didn't know where to ask the question.

Would it be possible to add a toSql function somewhere? I find myself needing to do that on a daily basis. The query builder is very powerful, but it can be confusing, so seeing the resulting sql helps a lot. I will submit a PR if you are willing to tell me where it could go.
